integrations/qdrant/src/haystack_integrations/document_stores/qdrant/converters.py

```diff
@@ -18,11 +18,21 @@ def documents_to_batch(
         documents: List[Document],
         *,
         embedding_field: str,
+        sparse_embedding_field: str,
     ) -> List[rest.PointStruct]:
         points = []
         for document in documents:
             payload = document.to_dict(flatten=False)
-            vector = payload.pop(embedding_field) or {}
+            vector = {}
+            if embedding_field in payload and payload[embedding_field] is not None:
+                dense_vector = payload.pop(embedding_field) or []
+                vector["text-dense"] = dense_vector
+            # TODO: does this work with this https://github.com/deepset-ai/haystack/pull/7382#discussion_r1530515595 ?
+            # TODO: maybe check if not empty string also on top of None ?
+            if sparse_embedding_field in payload and payload[sparse_embedding_field] is not None:
+                sparse_vector = payload.pop(sparse_embedding_field, {"indices": [], "values": []})
+                sparse_vector_instance = rest.SparseVector(**sparse_vector)
+                vector["text-sparse"] = sparse_vector_instance
```

Comment:
Lines 38 to 48 to be removed?

Reply:
the 
```python
            # TODO: remove as soon as we introduce the support for sparse embeddings in Qdrant
            if "sparse_embedding" in payload:
                sparse_embedding = payload.pop("sparse_embedding", None)
                if sparse_embedding:
                    logger.warning(
                        "Document %s has the `sparse_embedding` field set,"
                        "but storing sparse embeddings in Qdrant is not currently supported."
                        "The `sparse_embedding` field will be ignored.",
                        payload["id"],
                    )
```
I think yes but it was after my commits so I guess it's not in my PR @anakin87

Reply:
yes. I added these lines to not break Document Stores when we release the change to Document, but should be removed in this PR.